tests/js/session-tests.js

```diff
@@ -1152,6 +1152,225 @@ module.exports = {
+        in partition keys
+    */
+    const user3 = await app.logIn(credentials3);
+    const otherPartition = Utils.genPartition();
+    const config3 = {
+      sync: {
+        user: user3,
+        partitionValue: otherPartition,
+        _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+      },
+      schema: [schemas.PersonForSync, schemas.DogForSync],
+      path: realm3Path,
+    };
+
+    let realm3;
+    TestCase.assertThrowsAsyncContaining(async () => {
+      realm3 = await Realm.open(config3);
+    }, "Bad server version");
+
+    TestCase.assertUndefined(realm3);
+
+    realm1.close();
+  },
+
+  testRealmConversion: async function () {
+    const appConfig = AppConfig.integrationAppConfig;
+    let app = new Realm.App(appConfig);
+    let credentials = Realm.Credentials.anonymous();
+    let user = await app.logIn(credentials);
+
+    // local realm that we will put a few dogs into
+    const configLocal = {
+      schema: [schemas.PersonForSync, schemas.DogForSync],
+      path: "forSync.realm",
+    };
+    // a copy of the local realm -- still local
+    const configLocalCopy = {
+      schema: configLocal.schema,
+    };
+
+    // create five dogs in our local realm
+    const realmLocal = await Realm.open(configLocal);
+    realmLocal.write(() => {
+      for (let i = 0; i < 5; i++) {
+        realmLocal.create("Dog", {
+          _id: new ObjectId(),
+          breed: "breed",
+          name: "Brutus",
+          realm_id: "foo",
+        });
+      }
+    });
+
+    configLocalCopy.path = `${realmLocal.path}.copy.realm`;
+
+    /*  
+        Test 1:  invoke `writeCopyTo` with a Realm configuration rather than
+          explicit parameters.  Local -> local realm.
+    */
+    realmLocal.writeCopyTo(configLocalCopy);
+    const realmLocalCopy = await Realm.open(configLocalCopy);
+
+    /*  Test 2:  invoke `writeCopyTo` with a Realm configuration rather than
+          explicit parameters.  Local -> synced realm.
+    */
+    const loggedInUser = await app.logIn(credentials);
+    const configSync = {
+      path: `${realmLocal.path}.sync_copy.realm`,
+      sync: {
+        user: loggedInUser,
+        partitionValue: "foo",
+        _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+      },
+      schema: [schemas.DogForSync],
+    };
+
+    // create a copy of our realm that now has sync enabled
+    realmLocal.writeCopyTo(configSync);
+    const realmSynced = await Realm.open(configSync);
+    TestCase.assertDefined(realmSynced.syncSession);
+    await realmSynced.syncSession.uploadAllLocalChanges();
+    await realmSynced.syncSession.downloadAllServerChanges();
+
+    realmSynced.close();
+    realmLocalCopy.close();
+    realmLocal.close();
+  },
+
+  /*
+    Test the functionality of the writeCopyTo() operation on synced realms.
+  */
+  testSyncWriteCopyTo2: async function () {
```

Comment:
My suggestion would be to change this one to `testSyncWriteCopyToWithPath` and the other to `testSyncWriteCopyToWithConfig`